orchestrator/careplancontributor/service.go

```diff
@@ -40,11 +40,37 @@ func New(
 	fhirURL, _ := url.Parse(config.FHIR.BaseURL)
 	cpsURL, _ := url.Parse(config.CarePlanService.URL)
 
-	fhirClientConfig := coolfhir.Config()
-	localFHIRStoreTransport, _, err := coolfhir.NewAuthRoundTripper(config.FHIR, fhirClientConfig)
+	// Initialize FHIR clients:
+	// - FHIR API holding EHR data
+	// - FHIR API holding Questionnaires and HealthcareServices
+	localFhirStoreTransport, localFhirClient, err := coolfhir.NewAuthRoundTripper(config.FHIR, coolfhir.Config())
 	if err != nil {
 		return nil, err
 	}
+	var questionnaireFhirClient fhirclient.Client
+	if config.TaskFiller.QuestionnaireFHIR.BaseURL == "" {
+		// Questionnaire FHIR API not configured, use FHIR API holding EHR data
+		questionnaireFhirClient = localFhirClient
+	} else {
+		_, questionnaireFhirClient, err = coolfhir.NewAuthRoundTripper(config.TaskFiller.QuestionnaireFHIR, coolfhir.Config())
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Load Questionnaire-related resources for the Task Filler Engine from the configured URLs into the Questionnaire FHIR API
+	go func(ctx context.Context) {
+		if len(config.TaskFiller.QuestionnaireSyncURLs) > 0 {
+			log.Info().Ctx(ctx).Msgf("Synchronizing Task Filler Questionnaires resources from %d URLs", len(config.TaskFiller.QuestionnaireSyncURLs))
+			for _, u := range config.TaskFiller.QuestionnaireSyncURLs {
+				if err := coolfhir.ImportResources(ctx, questionnaireFhirClient, []string{"Questionnaire", "HealthcareService"}, u); err != nil {
+					log.Error().Ctx(ctx).Err(err).Msgf("Failed to synchronize Task Filler Questionnaire resources (url=%s)", u)
```

Comment:
I'm wondering, will this cause a deeper problem later if a flow tries to use a Questionnaire that could not be synced? 

Not to say we should crash the entire service if one fails to load, just thinking of edge cases.

I know it will fail out if the request comes in with an unsupported questionnaire, but it feels like a separate issue if a request fails for a questionnaire that should be supported but couldn't be loaded

Reply:
Well, it only needs to succeed once (until you update it of course). 

It could be a fact on the service health page? (which we don't have...). I added this as a thing we could implement as part of our monitoring efforts.